### DISCUSSION TITLE
Build pipeline question

### DISCUSSION THREAD

**Discussion by leeberryman** [wmgeolab/geoBoundaries#1703] [General]
@DanRunfola I'm curious about the bot / tests / and build script pipeline.  Are you spinning up your own vm with all of your specific dependencies for process?  Is it a GitHub action with a separate compute resource for each test with dependencies?  I'm currently processing a build pipeline and want to get rid of my arcpy dependency and I really like what you are doing with the automation, but I don't fully understand how the docker image is being used / deployed.

**Comment by DanRunfola** [comment]
Sure - it's github actions under the hood, coupled with a python script.

Most builds live here:
https://github.com/wmgeolab/geoBoundaries/tree/main/.github/workflows

And the image I use to run scripts lives here:
https://github.com/wmgeolab/geoBoundaries/tree/main/.github/workflows/workflowImages/ubuntu-small-gdal-plus

Taking this script as an example:
https://github.com/wmgeolab/geoBoundaries/blob/main/.github/workflows/fullBuild_gB_AFG_GRL.yml

You defined how it runs (in this case, a dispatch - means I click a button, but that'll transition to monthly automated soon).
Then you define the job.  Ignore the "strategy" for now - the runs-on tells it "use that image".

Under steps I define what happens.  The key take-away is I copy all of the files (git clone) out of the geoBoundaryBot repository, and then run the gbBuildCheck.py file.

That's really about it - obviously a lot of nuance, but that should hopefully get you started :)
